dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -152,4 +153,93 @@ private final FolderView getFolders(final Folder folder, final User user){
         return new FolderView(folder,foldersChildCustoms);
     }
 
+    /**
+     * This method will return a list of {@link FolderSearchResultView} that lives under
+     * the path or starts with the path sent.
+     *
+     * @param siteId site where the  folders should live under. If is not send will look under all hosts.
+     * @param pathToSearch path + folder name to find to
+     * @param user user
+     * @return list of {@link FolderSearchResultView}
+     */
+    public List<FolderSearchResultView> findSubFoldersPathByParentPath(final String siteId, final String pathToSearch, final User user)
+            throws DotSecurityException, DotDataException {
+        final List<FolderSearchResultView> subFolders = new ArrayList<>();
+
+        if(pathToSearch.lastIndexOf(StringPool.FORWARD_SLASH) == 0){ //If there is only one / we need to search the subfolders under the host(s)
+            if(UtilMethods.isSet(siteId)) {
+                subFolders.addAll(findSubfoldersUnderHost(siteId,pathToSearch,user));
+            } else{
+                final List<Host> siteList = APILocator.getHostAPI().findAll(user,false);
+                for(final Host site : siteList){
+                    if(subFolders.size() < 20) {
+                        subFolders.addAll(findSubfoldersUnderHost(site.getIdentifier(), pathToSearch, user));
+                    }
+                    continue;
+                }
+            }
+
+        } else {
+            if(UtilMethods.isSet(siteId)) {
+                subFolders.addAll(findSubfoldersUnderFolder(siteId,pathToSearch,user));
+            } else {
+                final List<Host> siteList = APILocator.getHostAPI().findAll(user,false);
+                for(final Host site : siteList){
+                    if(subFolders.size() < 20) {
+                        subFolders.addAll(findSubfoldersUnderFolder(site.getIdentifier(), pathToSearch, user));
+                    }
+                    continue;
+                }
+            }
+        }
+
+        return subFolders;
+    }
+
+    /**
+     * Will find the subfolders living directly under the host passed.
+     * If pathToSearch is sent is  gonna filter the path of the subfolder by it.
+     */
+    private List<FolderSearchResultView> findSubfoldersUnderHost(final String siteId,
+            final String pathToSearch, final User user)
+            throws DotSecurityException, DotDataException {
+        final List<FolderSearchResultView> subFolders = new ArrayList<>();
+
+        final Host host = APILocator.getHostAPI().find(siteId, user, false);
+        final List<Folder> subFoldersOfRootPath = APILocator.getFolderAPI()
+                .findSubFolders(host, user, false);
+        subFoldersOfRootPath.stream().filter(f -> f.getPath().startsWith(pathToSearch))
+                .limit(20).forEach(
```

Comment:
this should be config value

Reply:
rename f to folder

Reply:
Talked with Will about this, no right now.